src/util/performancetimer.h

```diff
@@ -1,75 +1,39 @@
-/****************************************************************************
-**
-** Copyright (C) 2012 Nokia Corporation and/or its subsidiary(-ies).
-** All rights reserved.
-** Contact: Nokia Corporation (qt-info@nokia.com)
-**
-** This file is part of the QtDeclarative module of the Qt Toolkit.
-**
-** $QT_BEGIN_LICENSE:LGPL$
-** GNU Lesser General Public License Usage
-** This file may be used under the terms of the GNU Lesser General Public
-** License version 2.1 as published by the Free Software Foundation and
-** appearing in the file LICENSE.LGPL included in the packaging of this
-** file. Please review the following information to ensure the GNU Lesser
-** General Public License version 2.1 requirements will be met:
-** http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html.
-**
-** In addition, as a special exception, Nokia gives you certain additional
-** rights. These rights are described in the Nokia Qt LGPL Exception
-** version 1.1, included in the file LGPL_EXCEPTION.txt in this package.
-**
-** GNU General Public License Usage
-** Alternatively, this file may be used under the terms of the GNU General
-** Public License version 3.0 as published by the Free Software Foundation
-** and appearing in the file LICENSE.GPL included in the packaging of this
-** file. Please review the following information to ensure the GNU General
-** Public License version 3.0 requirements will be met:
-** http://www.gnu.org/copyleft/gpl.html.
-**
-** Other Usage
-** Alternatively, this file may be used in accordance with the terms and
-** conditions contained in a signed written agreement between you and Nokia.
-**
-**
-**
-**
-**
-** $QT_END_LICENSE$
-**
-****************************************************************************/
-
 #pragma once
+#include <chrono>
 
-//
-// This is a fork of QPerformanceTimer just without the Q prefix
-// To fix interface changes issues in different QT versions
-// Added restart() function.
-// Returns time in nanosecond resolution.
-//
+#include "util/duration.h"
 
-#include <QtCore/qglobal.h>
+class PerformanceTimer {
+  public:
+    // TODO: make this configurable via a template parameter?
+    // note that the resolution of std::chrono::steady_clock is not guaranteed
+    // to be high resolution, but it is guaranteed to be monotonic.
+    // However, on all major platforms, it is high resolution enough.
+    using ClockT = std::chrono::steady_clock;
+    PerformanceTimer()
+            : m_startTime(ClockT::duration(0)){};
```

Comment:
Redundant
```suggestion
```

Reply:
Yes, technically redundant, but I wanted to make it explicit as well since we're relying on the fact thats its 0 in `PerformanceTimer::running`. Do you insist in the removal?